test/AutoDiff/validation-test/throw.swift

```diff
@@ -0,0 +1,83 @@
+// RUN: %target-run-simple-swift
+// REQUIRES: executable_test
+
+import StdlibUnittest
+import DifferentiationUnittest
+
+var ThrowingTests = TestSuite("Throwing")
+
+enum E: Error {
+    case error
+}
+
+ThrowingTests.testWithLeakChecking("SimpleTry") {
+  @differentiable(reverse)
+  func f(x: Double) throws -> Double {
+    if x < 0 {
+        throw E.error
+    }
+    else {
```

Comment:
Nit: probably, `} else {` is more preferable than `}` on one line and `else {` on the next one